libs/memory/include/hpx/memory/serialization/intrusive_ptr.hpp

```diff
@@ -0,0 +1,33 @@
+//  Copyright (c) 2014 Thomas Heller
```

Comment:
This looks identical to `libs/serialization/include/hpx/serialization/intrusive_ptr.hpp`. Shouldn't we keep only the one in the serialization module? This way we can use `intrusive_ptr` without depending on `serialization`.

Reply:
@msimber The one in the serialization module is for `boost::intrusive_ptr`. Leaving the serialization of `hpx::intrusive_ptr` there as well would make serialization depend on this module (and in the end on any other modules that requires serialization in any way). 

My rationale was to have the serialization module support only 'ambient' data types and to move all serialization of specific HPX types into their respective module. That makes those modules depend on `hpx_serialization`, but I thought that this is not a problem as serialization is a very fundamental functionality anyways. However I could see it being a problem in your case...

I have however excluded the header `memory/serialization/intrusive_ptr.hpp` from the generated module header. We could let this not depend on `hpx_serialization` and the user has to `#include <hpx/memory/serialization/intrusive_ptr.hpp>` explicitly anyways. So we could require for adding a dependency to `hpx_serialization` wherever this header is used as well.

Reply:
I'd see it the other way around. Serialization is an integral part and should be always a dependency. As you mentioned, it should be pretty lightweight now and shouldn't come with much bloat by itself, anything else leads to too much complexity for users.

Reply:
Well, that's what I said, didn't I? Maybe I mis-phrased things...

Reply:
Thanks @hkaiser. I was simply hoping we could avoid the dependency if possible, but since serialization is lightweight now let's do it like this!